lib/services/local/desktops.go

```diff
@@ -38,8 +38,61 @@ func NewWindowsDesktopService(backend backend.Backend) *WindowsDesktopService {
 	return &WindowsDesktopService{Backend: backend}
 }
 
+func (s *WindowsDesktopService) getWindowsDesktop(ctx context.Context, name, hostID string) (types.WindowsDesktop, error) {
+	key := backend.ExactKey(windowsDesktopsPrefix, hostID, name)
+	item, err := s.Get(ctx, key)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	desktop, err := s.itemToWindowsDesktop(item)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return desktop, nil
+}
+
+func (s *WindowsDesktopService) itemToWindowsDesktop(item *backend.Item) (types.WindowsDesktop, error) {
+	desktop, err := services.UnmarshalWindowsDesktop(
+		item.Value,
+		services.WithExpires(item.Expires),
+		services.WithRevision(item.Revision),
+	)
+	return desktop, trace.Wrap(err)
+}
+
+func (s *WindowsDesktopService) getWindowsDesktopsForHostID(ctx context.Context, hostID string) ([]types.WindowsDesktop, error) {
+	startKey := backend.ExactKey(windowsDesktopsPrefix, hostID)
+	result, err := s.GetRange(ctx, startKey, backend.RangeEnd(startKey), backend.NoLimit)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var desktops []types.WindowsDesktop
+	for _, item := range result.Items {
+		desktop, err := s.itemToWindowsDesktop(&item)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		desktops = append(desktops, desktop)
+	}
+
+	return desktops, nil
+}
+
 // GetWindowsDesktops returns all Windows desktops matching filter.
 func (s *WindowsDesktopService) GetWindowsDesktops(ctx context.Context, filter types.WindowsDesktopFilter) ([]types.WindowsDesktop, error) {
```

Comment:
Can we use the same implementation as `ListWindowsDesktops`, or are there differences?

We could keep the check on `req.Limit` in `ListWindowsDesktops` and delegate it and `GetWindowsDesktops` to an internal function that doesn't check, so we can give it an infinite limit for `GetWindowsDesktops`.

Reply:
The only difference right now is that when searching for a single desktop with name and host ID, `GetWindowsDesktops` returns a not found error and `ListWindowsDesktops` returns an empty result (but no error).

Reply:
wdyt of https://github.com/gravitational/teleport/commit/b3729c6f560391fc07f64b24851a87c838f69590 ?

Reply:
I don't hate it, though it's incomplete as it doesn't tolerate a limit of 0 and ends up truncating all the results.

Reply:
Oh yeah, I missed that 😬 

If we end up revisiting this (maybe as a result of the newly introduced backend streaming API?) we should definitely consider unifying the two functions.

Reply:
I will add a TODO before merging.